internal/gengapic/gengapic.go

```diff
@@ -380,9 +380,12 @@ func (g *generator) insertRequestHeaders(m *descriptor.MethodDescriptorProto, t
 		p("")
 		switch t {
 		case grpc:
-			p("ctx = insertMetadata(ctx, c.xGoogMetadata, md)")
+			p("hds = append(c.xGoogHeaders, hds...)")
+			p("ctx = gax.InsertMetadataIntoOutgoingContext(ctx, hds...)")
 		case rest:
-			p(`headers := buildHeaders(ctx, c.xGoogMetadata, md, metadata.Pairs("Content-Type", "application/json"))`)
+			p(`hds = append(c.xGoogHeaders, hds...)`)
+			p(`hds = append(hds, "Content-Type", "application/json")`)
```

Comment:
I kind of wish `"Content-Type", "application/json"` was just part of `c.xGoogHeaders` ya know? It's set on every regapic request...but `xGoogHeaders` has always been `x-goog-api-client` headers only...idk wdyt? @codyoss hbu?

Reply:
I'll wait to hear back from Cody on this before merging.

Reply:
I would be fine with it just being agnostic and just general headers.

Reply:
I would imagine `XGoogHeaders` is specifically for headers with the x-goog prefix (of which there are many, not just `x-goog-api-client`)

Reply:
@noahdietz What's your call on this?

Reply:
Let's just keep it as is.